test/ha/ha.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ha
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	pkgTest "knative.dev/pkg/test"
+	rtesting "knative.dev/serving/pkg/testing/v1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+	v1test "knative.dev/serving/test/v1"
+)
+
+const (
+	servingNamespace = "knative-serving"
+	haReplicas       = 2
+)
+
+func getLeader(t *testing.T, clients *test.Clients, lease, labelSelector string) (string, error) {
+	var leader string
+	if err := wait.PollImmediate(test.PollInterval, time.Minute, func() (bool, error) {
+		lease, err := clients.KubeClient.Kube.CoordinationV1().Leases(servingNamespace).Get(lease, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting lease %s: %w", lease, err)
+		}
+		leader = strings.Split(*lease.Spec.HolderIdentity, "_")[0]
+		currentPodList, err := clients.KubeClient.Kube.CoreV1().Pods(servingNamespace).List(metav1.ListOptions{
```

Comment:
what about just `Get` that pod?

Reply:
There can be more pods. I want to be able to change the constant haReplicas and still be able to run this test.

Reply:
Maybe I could set haReplicas to 3 for the test as a default? I wanted to minimize the number of pods that have to be started but 3 pods should be fine as well.

Reply:
Sure there can be more. But only 1 that holds the lease, no?

Reply:
OK. I get your point now. Adding one more commit to simplify the stuff.